sklearn/utils/estimator_checks.py

```diff
@@ -1086,7 +1090,7 @@ def check_sample_weights_shape(name, estimator_orig):
 
 
 @ignore_warnings(category=FutureWarning)
-def check_sample_weight_equivalence(name, estimator_orig):
+def _check_sample_weight_equivalence(name, estimator_orig, sparse_container):
```

Comment:
```suggestion
def _check_sample_weight_equivalence(name, estimator_orig, sparse_container=None):
```

Reply:
I wonder if it wouldn't be better to have two functions for the dense and sparse inputs:

`check_sample_weight_equivalence_on_dense_data` and `check_sample_weight_equivalence_on_sparse_data`.

This way, we could configure different hyperparameter configs in `PER_ESTIMATOR_CHECK_PARAMS` if needed.

On the other hand, this might introduce a lot of redundant entries in that dict.

Any opinion @adrinjalali?

Reply:
yeah we had another case where a parameter would change the behavior of the test and it made sense to create two tests for them. I think it makes sense to try to keep the signature of the tests to `name, estimator_orig` and the rest should be defined by the test itself.

Reply:
I think the two distinct `check_sample_weight_equivalence_on_dense_data` and `check_sample_weight_equivalence_on_sparse_data` names would be more convenient for filtering with the -k flag in pytest (we can then easily do only dense, only sparse, or both).